app/controllers/api/person_escort_records_controller.rb

```diff
@@ -35,12 +35,14 @@ def assessment_serializer
       PersonEscortRecordSerializer
     end
 
-    def create_confirmation_event!
+    def create_confirmation_event_and_notification!
       create_automatic_event!(eventable: assessment, event_class: GenericEvent::PerConfirmation, details: { confirmed_at: assessment.confirmed_at.iso8601 })
+      Notifier.prepare_notifications(topic: assessment, action_name: nil)
```

Comment:
could you give `action_name`  a value rather than be nil - its useful to see the reason for the notification and it is required in some of the emails

Reply:
ah ok, I see the action_name is being set further down the chain in `PrepareAssessmentNotificationsJob`. I guess there is a reason for setting action_name there, rather than up here?

Reply:
Hmm... this isn't trivial to change. Previously the behaviour was shared across both assessment types and the `action_name` parameter was passed as simply `update`. I've refactored as the behaviour for each assessment has diverged but essentially for the existing notifications the same behaviour is preserved. I did find that passing a string value that was effectively ignored within the `Notifier` class was a little misleading, but as `action_name` is a required parameter I changed this to `nil`.

It would be possible to pass the `action_name` value as either `confirm_person_escort_record` or `confirm_youth_risk_assessment` rather than deriving this within `PrepareAssessmentNotificationsJob` but that would mean changing the job `perform` signature and potentially breaking any current queued jobs. So that in turn means adding a new class definition for the new method signature, deploying that and in a further PR removing the old job once we're sure there are no jobs queued.

All of which feels beyond the scope of this PR unfortunately so for now I've just documented with a `TODO` to resolve in future.

```ruby
class PrepareAssessmentNotificationsJob < ApplicationJob
  include QueueDeterminer

  def perform(topic_id:, topic_class:, **_)
    assessment = topic_class.constantize.find(topic_id)
    return unless (move = assessment.move)

    PrepareMoveNotificationsJob.perform_now(topic_id: move.id, action_name: "confirm_#{topic_class.underscore}", queue_as: move_queue_priority(move))
  end
end
```

class PrepareAssessmentNotificationsJob < ApplicationJob
  include QueueDeterminer

  def perform(topic_id:, topic_class:, **_)
    assessment = topic_class.constantize.find(topic_id)
    return unless (move = assessment.move)

    PrepareMoveNotificationsJob.perform_now(topic_id: move.id, action_name: "confirm_#{topic_class.underscore}", queue_as: move_queue_priority(move))
  end
end

Reply:
ok, understood, thanks 👍